dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -101,33 +101,59 @@ def manual_pass
     redirect_to :manual_pass_form
   end
 
+  # display the user search form, users found, and permissions for first user found
+  # grant / revoke permissions actions redirect back to this form to display the updated permission list
   def permissions_form
+    @users = Array.new # list of users that match search
+    @permissions = Array.new
+    search_term = user_permission_params[:search_term]
+    if search_term =~ /^\d+$/
+      user_id = search_term
+      @users = User.where(id: user_id)
+    elsif search_term
+      email = search_term
+      hashed_email = User.hash_email(email)
+      @users = User.where(hashed_email: hashed_email)
+    end
+    @user = @users.first if @users
+    @permissions = @user.permissions if @user
   end
 
-  # Grants the indicated permission to the indicated user. Expects params[:email] and
-  # params[:permission] to be populated.
   def grant_permission
-    permission = params[:permission]
-    user = User.find_by_email_or_hashed_email params[:email]
+    user_id = user_permission_params[:user_id]
+    permission = user_permission_params[:user_permission_id]
+    @user = User.find(user_id)
 
-    unless user && user.teacher?
-      flash[:alert] = "FAILED: user #{params[:email]} could not be found or was not a teacher"
-      redirect_to :permissions_form
+    unless @user && @user.teacher?
+      flash[:alert] = "FAILED: user #{user_permission_params[:user_id]} could not be found or was not a teacher"
+      redirect_to action: "permissions_form", search_term: user_id
       return
     end
 
     if permission == 'admin'
-      if user.sections_as_student.count > 0
-        flash[:alert] = "FAILED: user #{user.email} NOT granted as user has sections_as_students"
+      if @user.sections_as_student.count > 0
```

Comment:
It appears that User model does not use the `:strict` option when validating section count.  It just adds ActiveModel::Errors which, I believe, don't raise exceptions.

``` ruby
  validate :admins_must_be_teachers_without_followeds

  def admins_must_be_teachers_without_followeds
    if admin
      errors.add(:admin, 'must be a teacher') unless teacher?
      errors.add(:admin, 'cannot be a followed') unless sections_as_student.empty?
    end
  end
```

Should I add the `:strict` option to this validation so that it raises `ActiveModel::StrictValidationFailed` (which sub-classes `StandardError` and is an exception that can be rescued)?  That would eliminate the duplicate validation logic in the controller.